addons/web/static/src/core/effects/rainbow_man.scss

```diff
@@ -1,256 +1,156 @@
-  @include media-breakpoint-down(sm) {
-    margin: -5% auto 0 (-$reward-size-mobile / 2);
-  }
-  background-image: -webkit-radial-gradient(#edeff4 30%, transparent 70%, transparent);
-  background-image: -o-radial-gradient(#edeff4 30%, transparent 70%, transparent);
-  background-image: radial-gradient(#edeff4 30%, transparent 70%, transparent);
-
-  animation: reward-fading ($reward-base-time * 0.5) ease-in-out 0s 1 normal forwards;
-  @include o-position-absolute(20%, auto, auto, 50%);
-  @include size($reward-size, $reward-size);
-  @include media-breakpoint-down(sm) {
-    @include size($reward-size-mobile, $reward-size-mobile);
-  }
-
-  &.o_reward_fading {
-    display: block;
-    animation: reward-fading-reverse ($reward-base-time * 0.4) ease-in-out 0s 1 normal forwards;
+    inset: 0; // Same than : left: 0; right: 0; top: 0; bottom: 0;
+    will-change: transform;
+    z-index: $zindex-modal;
+    animation: reward-fading $reward-base-time * 0.5 ease-in-out forwards;
 
-    .o_reward_face_group {
-      animation: reward-jump-reverse ($reward-base-time * 0.4) ease-in-out 0s 1 normal forwards;
+    .o_reward_box {
+        transform-box: fill-box;
     }
-
-    .o_reward_rainbow {
-      path {
-        animation: reward-rainbow-reverse ($reward-base-time * 0.5) ease-out 0s 1 normal forwards;
-      }
+    &.o_reward_fading {
+        animation: reward-fading-reverse $reward-base-time * 0.4 ease-in-out forwards;
+
+        .o_reward_face_group {
+            animation: reward-jump-reverse $reward-base-time * 0.4 ease-in-out forwards;
+        }
+        .o_reward_rainbow_line {
+            animation: reward-rainbow-reverse $reward-base-time * 0.5 ease-out forwards;
+        }
     }
-  }
-
-  .o_reward_face,
-  .o_reward_stars,
-  .o_reward_shadow,
-  .o_reward_thumbup {
-    margin: 0 auto;
-  }
-
-  .o_reward_rainbow {
-    path {
-      stroke-dasharray: 600;
-      stroke-dashoffset: 0;
-      fill: none;
-      stroke-linecap: round;
-      stroke-width: 21px;
-      animation: reward-rainbow $reward-base-time ease-out 0s 1 normal forwards;
-    }
-  }
-
-  .o_reward_face_group {
-    transform-origin: center;
-    animation: reward-jump $reward-base-time * 0.8 ease-in-out 0s 1 normal none running;
-    @include o-position-absolute(6%, 0, 0, 0);
-    @include size(100%, 60%);
-  }
-
-  .o_reward_face {
-    display: block;
-    top: 42%;
-    position: relative;
-    border-radius: 100%;
-    background: center center / contain no-repeat;
-    animation: reward-float $reward-base-time ease-in-out $reward-base-time infinite alternate;
-    @include size(34%, 56.67%);
-  }
-
-  .o_reward_stars {
-    display: block;
-    @include size($reward-size * 0.75, $reward-size / 2);
-    @include media-breakpoint-down(sm) {
-      @include size($reward-size-mobile * 0.75, $reward-size-mobile / 2);
+    .o_reward_rainbow_man,
+    foreignObject {
+        // Could be replaced by the "overflow-visible" class after the migration to BS5
```

Comment:
define "overflow-visible" in `web\...\webclient.scss` directly.
Add a note saying that the class can be removed with v5 (like  "Inline Margins")